src/frontend/src/binder/relation/table_or_source.rs

```diff
@@ -107,25 +109,29 @@ impl Binder {
             match schema_name {
                 Some(schema_name) => {
                     let schema_path = SchemaPath::Name(schema_name);
-                    if schema_name == PG_CATALOG_SCHEMA_NAME {
-                        if let Ok(sys_table_catalog) = self
-                            .catalog
-                            .get_sys_table_by_name(&self.db_name, table_name)
-                        {
+                    if schema_name == PG_CATALOG_SCHEMA_NAME
```

Comment:
How about providing a function like `is_system_schema` to check schema name here? So we can reuse it in L163 and handle some other system schemas in that function as well in the future maybe.